QHyper/problems/TSP.py

```diff
@@ -0,0 +1,135 @@
+import itertools
+
+import numpy as np
+
+from .problem import Problem
+
+
+class TSP:
+    """TSP class"""
+
+    def __init__(self, number_of_cities: int, coords_range: tuple[float, float]=(0, 10000)) -> None:
+        self.number_of_cities = number_of_cities
+        self.coords_range = coords_range
+        self.cities_coords = self.get_cities()
```

Comment:
This relates to my question about defining attributes outside of init in the knapsack class - maybe this is the wrong way to do it? Have you ever found any resources about that?